src/torch_ucc.cpp

```diff
@@ -236,12 +239,31 @@ std::shared_ptr<CommPG> CommPG::get_comm(
   static uint32_t comm_id;
 
   std::lock_guard<std::mutex> lock(m);
-  id = (comm_id++ % TORCH_UCX_COMM_BITS);
-  oob->comm_id = id;
+  id = (comm_id % TORCH_UCX_COMM_BITS);
```

Comment:
copy Pallab's comment (cc @pallab-zz)

>  this should be TORCH_UCX_MAX_COMM

```suggestion
  id = (comm_id % TORCH_UCX_MAX_COMM);
```

Reply:
good catch